src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs

```diff
@@ -42,8 +46,150 @@ public void Handle(List<PostProcessor> postProcessors, Manifest manifest, string
+                    _increContext.CurrentInfo.PostProcessOutputs.Add(outputRelPath, currentCachedFileName);
+                });
+            }
+        }
+
+        private void TraceNoneIncremental(string outputFolder, List<ManifestItem> noneIncreItems)
+        {
+            foreach (var outputRelPath in GetOutputRelativePaths(noneIncreItems))
+            {
+                string lastCachedRelPath;
+                if (_increContext.LastInfo != null && _increContext.LastInfo.PostProcessOutputs.TryGetValue(outputRelPath, out lastCachedRelPath))
+                {
+                    throw new BuildCacheException($"Last incremental post processor outputs shuold not contain {outputRelPath}.");
+                }
+
+                IncrementalUtility.RetryIO(() =>
+                {
+                    var outputPath = Path.Combine(outputFolder, outputRelPath);
+                    var currentCachedFileName = IncrementalUtility.GetRandomEntry(_increContext.CurrentBaseDir);
+
+                    // Copy output to current cached file
+                    File.Copy(outputPath, Path.Combine(_increContext.CurrentBaseDir, currentCachedFileName));
+                    _increContext.CurrentInfo.PostProcessOutputs.Add(outputRelPath, currentCachedFileName);
+                });
+            }
+        }
+
+        #endregion
+
+        private void CopyToOutput(List<ManifestItem> increItems, string outputFolder)
+        {
+            foreach (var outputRelPath in GetOutputRelativePaths(increItems))
+            {
+                string lastCachedRelPath;
+                if (!_increContext.LastInfo.PostProcessOutputs.TryGetValue(outputRelPath, out lastCachedRelPath))
+                {
+                    throw new BuildCacheException($"Last incremental post processor outputs shuold contain {outputRelPath}.");
+                }
+
+                IncrementalUtility.RetryIO(() =>
+                {
+                    // Copy last cached file to output
+                    var outputPath = Path.Combine(outputFolder, outputRelPath);
+                    var lastCachedFile = Path.Combine(_increContext.LastBaseDir, lastCachedRelPath);
+                    Directory.CreateDirectory(Path.GetDirectoryName(outputPath));
+                    File.Copy(lastCachedFile, outputPath, true);
+                });
+            }
+        }
+
+        private static IEnumerable<string> GetOutputRelativePaths(List<ManifestItem> items)
+        {
+            return from item in items
+                   from output in item.OutputFiles.Values
+                select output.RelativePath;
+        }
+
+        private static void CheckNoIncrementalItems(Manifest manifest)
+        {
+            if (manifest.Files.Any(i => i.IsIncremental))
+            {
+                throw new DocfxException("Manifest items are not supposed to have any incremental items.");
```

Comment:
yes should be more specific

---
In reply to: [95292970](https://github.com/dotnet/docfx/pull/1136#discussion_r95292970) [](ancestors = 95292970)